### ISSUE TITLE
Intellij sbt shell repeatedly warns of "Unrecognized repository"

### ISSUE THREAD

**Issue by ashleymercer** [scalacenter/scalafix#1887]
I have a multi-project sbt build and I've recently added some custom scalafix rules in-tree using the [instructions here](https://scalacenter.github.io/scalafix/docs/developers/local-rules.html#multi-project-sbt-build)

When I run any scalafix commands in IntelliJ's sbt shell, I get hundreds (no exaggeration) of warnings that all say
(EDIT: actually this happens even when running unrelated commands like `compile` or `test`)

```
[warn] Unrecognized repository Scala Plugin Bundled Repository, ignoring it
```

This is mildly frustrating since it completely swamps the "real" output of the shell and makes it hard to find any actual problems.

**Comment by tgodzik** [comment]
Thanks for reporting! Did you try asking on the Scala plugin discord or reporting an issue with Jetbrains? 

The discord link is here https://discord.gg/aUKpZzeHCK

**Comment by ashleymercer** [comment]
Ahh yes having done some more digging - this only happens in IntelliJ's sbt shell, and not when using regular command-line sbt. I've cross-posted to their bug tracker:
https://youtrack.jetbrains.com/issue/IDEA-337146/sbt-projects-using-scalafix-repeatedly-warn-of-Unrecognized-repository

**Comment by bjaglin** [comment]
Per [this commit](https://github.com/JetBrains/intellij-scala/commit/cced397730039519815943746ce4efca3ddcadc0), it looks like the issue arise when the IntelliJ sbt plugin is injected into a project using sbt-coursier, and is fixed in recent 2023.3.x IntelliJ versions.

**Comment by bjaglin** [comment]
I confirm that users of IntelliJ < 2023.3.2 running projects with `ScalafixTestkitPlugin` are impacted through https://github.com/scalacenter/sbt-scalafix/blob/8d817d76cafa82ae31b70c7d69f8af4c9fa04e74/src/main/scala/scalafix/sbt/ScalafixTestkitPlugin.scala#L37.

I am still trying to figure out what's going on. Until then, this workaround seems to work for https://github.com/ashleymercer/scalafix-issue-1887:
```scala
ThisBuild / includePluginResolvers := false
```

**Comment by bjaglin** [comment]
> I am still trying to figure out what's going on.

* it looks like there are different code paths for the build and the metabuild (coursier vs ivy?)
* by construction, `includePluginResolvers := true` exposes the plugin resolver added by the IntelliJ sbt plugin to a code path not exercised in the default case: the build one, driven by lm-coursier-shaded as published by sbt-coursier
* the plugin resolver, as declared before IntelliJ 2023.3.2, was not valid for that code path, generating the warning in all `update` in the build (without any impact as the resolver is only needed in the metabuild)

I'll close this as this is fixed in the latest IntelliJ version, and there is a workaround above for the older ones. Thanks again for the report, feel free to reopen if my analysis is wrong.

**Comment by ashleymercer** [comment]
Awesome detective work @bjaglin - thanks!

Yes can confirm this is fixed in the EAP builds of 2023.3 so happy for this to be closed.
